src/sentry/static/sentry/app/views/settings/components/forms/choiceMapper.jsx

```diff
@@ -0,0 +1,188 @@
+import {Flex, Box} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import {defined, objectIsEmpty} from 'app/utils';
+import {t} from 'app/locale';
+import Button from 'app/components/buttons/button';
+import DropdownAutoComplete from 'app/components/dropdownAutoComplete';
+import DropdownButton from 'app/components/dropdownButton';
+import InputField from 'app/views/settings/components/forms/inputField';
+import SelectControl from 'app/components/forms/selectControl';
+
+export default class ChoiceMapper extends React.Component {
+  static propTypes = {
+    ...InputField.propTypes,
+    /**
+     * Text used for the 'add row' button.
+     */
+    addButtonText: PropTypes.node,
+    /**
+     * Configuration for the add item dropdown.
+     */
+    addDropdown: PropTypes.shape(DropdownAutoComplete.propTypes).isRequired,
+    /**
+     * The label to show above the row name selected from the dropdown.
+     */
+    mappedColumnLabel: PropTypes.node,
+    /**
+     * A list of column labels (headers) for the multichoice table. This should
+     * have the same number of items as the mappedSelectors prop.
+     */
+    columnLabels: PropTypes.objectOf(PropTypes.node).isRequired,
+    /**
+     * A list of select field properties that should be used to render the
+     * select field for each column in the row.
+     */
+    mappedSelectors: PropTypes.objectOf(PropTypes.objectOf(SelectControl.propTypes))
```

Comment:
I would guess we will eventually want to support any fields (prob by using `FieldFromConfig`.

Reply:
This might be a little tricky since the fields will think their bound to a form, since it will have that context. We can think on that later when we actually need it to do that